Sources/Starknet/Network/HttpNetworkProvider.swift

```diff
@@ -59,12 +65,17 @@ class HttpNetworkProvider {
             throw HttpNetworkProviderError.unknownError
         }
 
-        if let result = try? JSONDecoder().decode(U.self, from: data) {
+        do {
+            let result = try JSONDecoder().decode(U.self, from: data)
             return result
-        } else if let response = response as? HTTPURLResponse, response.statusCode < 200 || response.statusCode > 299 {
-            throw HttpNetworkProviderError.requestRejected
+        } catch {
+            if let response = response as? HTTPURLResponse, response.statusCode < 200 || response.statusCode > 299 {
+                throw HttpNetworkProviderError.requestRejected
+            } else if let decodingError = error as? DecodingError {
+                throw HttpNetworkProviderError.decodingError(decodingError)
+            } else {
+                throw HttpNetworkProviderError.unknownError
+            }
```

Comment:
This adresses #112 

For now, to keep errors somewhat abstract, I've added intermediate `.decodingError(DecodingError)` and `.encodingError(EncodingError)`.

We could also use approach used by `StarknetProvider`, and transform those to:
```swift
enum HttpNetworkProviderError: Error {
case encodingError(String)
case encodingError(String)
...
}
```
Then throw in one of these ways:
```swift
throw HttpNetworkProviderError.decodingError(decodingError.localizedDescription)
throw HttpNetworkProviderError.decodingError(decodingError.localizedDescription.description)
throw HttpNetworkProviderError.decodingError(decodingError.localizedDescription.debugDescription)
```
This approach, however, will provide shorter and less self-explanatory error messages. 

For now I opted for the first approach, as it provides full "stack-trace" kind-of errors that help quickly localize the issue. 

The question is if it's okay to deviate from the previously used approach. wdyt?

Reply:
In my opinion this is ok, as explicit is better than implicit.